src/Operators/pointwisestencil.jl

```diff
@@ -337,23 +360,75 @@ get_boundary(
 stencil_interior_width(::PointwiseStencilOperator, stencil, arg) =
     ((0, 0), bandwidths(eltype(stencil)))
 
-function boundary_width(
-    ::PointwiseStencilOperator,
-    ::LeftStencilBoundary,
+function left_interior_idx(
+    space::Union{
+        Spaces.FaceFiniteDifferenceSpace,
+        Spaces.FaceExtrudedFiniteDifferenceSpace,
+    },
+    op::PointwiseStencilOperator,
+    bc::LeftStencilBoundary,
+    stencil,
+    arg,
+)
+    lbw = bandwidths(eltype(stencil))[1]
+    if lbw isa Integer
+        return left_idx(space) - lbw
+    elseif lbw isa PlusHalf
+        return left_idx(space) - lbw + half
+    end
+end
+function left_interior_idx(
+    space::Union{
+        Spaces.CenterFiniteDifferenceSpace,
+        Spaces.CenterExtrudedFiniteDifferenceSpace,
+    },
+    op::PointwiseStencilOperator,
+    bc::LeftStencilBoundary,
     stencil,
     arg,
 )
     lbw = bandwidths(eltype(stencil))[1]
-    return max((left_idx(axes(arg)) - lbw) - left_idx(axes(stencil)), 0)
+    if lbw isa Integer
+        return left_idx(space) - lbw
+    elseif lbw isa PlusHalf
+        return left_idx(space) - lbw - half
+    end
 end
-function boundary_width(
-    ::PointwiseStencilOperator,
-    ::RightStencilBoundary,
+
+
+@inline function right_interior_idx(
+    space::Union{
+        Spaces.FaceFiniteDifferenceSpace,
+        Spaces.FaceExtrudedFiniteDifferenceSpace,
+    },
+    op::PointwiseStencilOperator,
+    bc::RightStencilBoundary,
     stencil,
     arg,
 )
-    ubw = bandwidths(eltype(stencil))[2]
-    return max(right_idx(axes(stencil)) - (right_idx(axes(arg)) - ubw), 0)
+    rbw = bandwidths(eltype(stencil))[2]
+    if rbw isa Integer
+        return right_idx(space) - rbw
+    elseif rbw isa PlusHalf
+        return right_idx(space) - rbw - half
+    end
+end
+@inline function right_interior_idx(
+    space::Union{
+        Spaces.CenterFiniteDifferenceSpace,
+        Spaces.CenterExtrudedFiniteDifferenceSpace,
+    },
+    op::PointwiseStencilOperator,
+    bc::RightStencilBoundary,
+    stencil,
+    arg,
+)
+    rbw = bandwidths(eltype(stencil))[2]
+    if rbw isa Integer
+        return right_idx(space) - rbw
+    elseif rbw isa PlusHalf
+        return right_idx(space) - rbw + half
+    end
```

Comment:
Should we define

```julia
right_interior_idx(rbw::Integer, space) = right_idx(space) - rbw
right_interior_idx(rbw::PlusHalf, space) = right_idx(space) - rbw + half
```
and use that here and above? Actually, it looks like the two `right_interior_idx` can be unified with these changes?

Reply:
Good idea.